command/agent/agent_endpoint.go

```diff
@@ -332,8 +332,15 @@ func (s *HTTPServer) AgentForceLeaveRequest(resp http.ResponseWriter, req *http.
 		return nil, CodedError(400, "missing node to force leave")
 	}
 
+	_, prune := req.URL.Query()["prune"]
+
 	// Attempt remove
-	err := srv.RemoveFailedNode(node)
+	var err error
+	if prune {
+		err = srv.RemoveFailedNodePrune(node)
+	} else {
+		err = srv.RemoveFailedNode(node)
+	}
 	return nil, err
```

Comment:
This sets `prune` to `true` even if the `prune` query parameter is `?prune=false`, because it's only checking for the _existence_ of the key, for which the value is `"false"` (a string, not a bool). 

You should use the `parseBool` helper in `command/http.go` rather than the stdlib function, as it returns (`nil, nil`) if the parameter isn't present and you can have that default to the existing behavior as you expect.

Reply:
Thanks so much! This has been updated.